app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/AddFriend.kt

```diff
@@ -0,0 +1,218 @@
+import com.github.se.gomeet.ui.mainscreens.events.GoMeetSearchBar
+import com.github.se.gomeet.ui.navigation.NavigationActions
+import com.github.se.gomeet.ui.navigation.Route
+import com.github.se.gomeet.viewmodel.UserViewModel
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AddFriend(nav: NavigationActions, userViewModel: UserViewModel) {
+
+  val screenWidth = LocalConfiguration.current.screenWidthDp.dp
+  val screenHeight = LocalConfiguration.current.screenHeightDp.dp
+  val query = remember { mutableStateOf("") }
+  var isLoading by remember { mutableStateOf(true) }
+  var users by remember { mutableStateOf(listOf<GoMeetUser>()) }
+  var followedUsers by remember { mutableStateOf(setOf<String>()) }
+
+  val coroutineScope = rememberCoroutineScope()
+
+  LaunchedEffect(Unit) {
+    coroutineScope.launch {
+      val fetchedUsers = userViewModel.getAllUsers()
+      val currentUser = Firebase.auth.currentUser!!.uid
+      if (fetchedUsers != null) {
+        users = fetchedUsers.filter { it.uid != currentUser } // Exclude the current user
+        // Initialize the followed users
+        val currentUserData = userViewModel.getUser(currentUser)
+        if (currentUserData != null) {
+          followedUsers = currentUserData.following.toSet()
+        }
+      }
+      isLoading = false
+    }
+  }
+
+  // Filtered users based on search query
+  val filteredUsers by
+      remember(query.value, users) {
+        derivedStateOf {
+          if (query.value.isEmpty()) {
+            users
+          } else {
+            users.filter { it.username.contains(query.value, ignoreCase = true) }
+          }
+        }
+      }
+
+  Scaffold(
+      topBar = {
+        Column {
+          TopAppBar(
+              modifier = Modifier.testTag("TopBar"),
+              backgroundColor = MaterialTheme.colorScheme.background,
+              elevation = 0.dp,
+              title = {
+                // Empty title since we're placing our own components
+              },
+              navigationIcon = {
+                IconButton(onClick = { nav.goBack() }) {
+                  Icon(
+                      imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                      contentDescription = "Back",
+                      tint = MaterialTheme.colorScheme.onBackground)
+                }
+              })
+
+          Row(
+              verticalAlignment = Alignment.CenterVertically,
+              modifier = Modifier.padding(start = screenWidth / 15, top = 0.dp)) {
+                Text(
+                    text = "Find User",
+                    style =
+                        MaterialTheme.typography.headlineMedium.copy(
+                            fontWeight = FontWeight.SemiBold))
+              }
```

Comment:
Text might look better : smaller and on the same horizontal height as the goBack Button.
What are your thoughts on this change ?

Reply:
It does look better, thanks!